src/UI.elm

```diff
@@ -0,0 +1,34 @@
+module UI exposing (showBoard)
+
+import Html exposing (Html, div, text, button)
+import Html.App as Html
+import Board exposing (Board)
+import Array exposing (Array, length, map, initialize, slice, toList)
+
+showBoard : Board -> Html a
+showBoard board =
+  div [] (showRows board)
+
+
+showRows : Board -> List (Html a)
+showRows board =
+  (create2DBoard board)
+    |> List.map (\line -> div [] (showCells line)) 
+
+
+showCells line =
+  line
+    |> List.map (\cell -> button [] [(text cell)])
+
+
+create2DBoard : Board -> List (List String)
+create2DBoard board =
+  let
+      width = round (sqrt (toFloat (length board)))
```

Comment:
Hmmm. If I moved the create2DBoard function to function within board called `getBoard`. Would that be sufficient. But then the `showRows` and `showCells` functions will still know about it cause they are mapping over it.